src/UiPath.Workflow/Activities/RoslynExpressionValidator.cs

```diff
@@ -31,62 +37,35 @@ public abstract class RoslynExpressionValidator
     /// </param>
     protected RoslynExpressionValidator(HashSet<Assembly> seedAssemblies = null)
     {
-        _metadataReferenceDictionary = new(GetInitialMetadataReferences);
-        var assembliesToReference = new HashSet<Assembly>();
-        assembliesToReference.UnionWith(JitCompilerHelper.DefaultReferencedAssemblies);
+        _metadataReferences = new(GetInitialMetadataReferences);
+
+        var assembliesToReference = new HashSet<Assembly>(JitCompilerHelper.DefaultReferencedAssemblies);
         if (seedAssemblies != null)
         {
-            assembliesToReference.UnionWith(seedAssemblies);
+            assembliesToReference.UnionWith(seedAssemblies.Where(a => a is not null));
         }
 
-        assembliesToReference.RemoveWhere(x => x == null);
         RequiredAssemblies = assembliesToReference;
     }
 
     /// <summary>
     ///     Assemblies required on the <see cref="Compilation"/> object. Use <see cref="AddRequiredAssembly(Assembly)"/>
     ///     to add more assemblies.
     /// </summary>
-    protected IReadOnlySet<Assembly> RequiredAssemblies { get; private set; }
+    protected IReadOnlySet<Assembly> RequiredAssemblies { get; init; }
 
     /// <summary>
     ///     The kind of identifier to look for in the syntax tree as variables that need to be resolved for the expression.
     /// </summary>
     protected abstract int IdentifierKind { get; }
 
-    /// <summary>
-    ///     Adds an assembly to the <see cref="RequiredAssemblies"/> set.
-    /// </summary>
-    /// <param name="assembly">assembly</param>
-    /// <remarks>
-    ///     Takes a lock and replaces <see cref="RequiredAssemblies"/> with a new set. Lock is taken in case
-    ///     multiple threads are adding assemblies simultaneously.
-    /// </remarks>
-    public void AddRequiredAssembly(Assembly assembly)
```

Comment:
This is used in Studio Web and it's public surface area. Is there a reason to remove it entirely?

Reply:
That's my bad, I didn't realize it's part of the public API. I've restored it.